src/motioncore/protocols/constant/constant_share_wrapper.h

```diff
@@ -0,0 +1,114 @@
+// MIT License
+//
+// Copyright (c) 2022 Liang Zhao
+// Cryptography and Privacy Engineering Group (ENCRYPTO)
+// TU Darmstadt, Germany
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#pragma once
+
+#include "protocols/share_wrapper.h"
+#include "utility/meta.hpp"
+
+namespace encrypto::motion {
+
+/// \brief class to wrap constant share input functions
```

Comment:
Is it possible to use `ConstantShareWrapper `s functionality in `ShareWrapper`, so without adding an extra class? I don't understand its purpose right now.

Reply:
I use `ConstantShareWrapper` simply to make the creation of constant shares easier (or cleaner). 
The use of `ConstantShareWrapper` can be completely avoided by using functions in `Backend.cpp` and `BitVector.cpp`. 
But if so, when we later implement floating-point arithmetic (arithmetic GMW), we may have to call the functions in `Backend.cpp` and `BitVector.cpp` for many times rather than `ConstantShareWrapper`.

Which method would you recommend?

Reply:
Maybe use something like `Party::PublicIn` and call `Party::SharedIn` internally?

Reply:
I should move the functions of `ConstantShareWrapper` to `ShareWrapper`. 
But `Party::SharedIn` seems not able to solve the problems (please see [discussion](https://github.com/encryptogroup/MOTION/pull/29#discussion_r956728302)).